### PR TITLE
Cloud download

### PR DESCRIPTION
**Pull request by alexisandreason**

### CODE REVIEW

File: google_cloud_storage/index.js [hunk 1]
```diff
@@ -23,5 +28,42 @@ async function uploadFile(input) {
...
+
+    const storage = new Storage({
+        projectId: input['ProjectId'],
+        credentials: credential,
+    });
+    const bucket = storage.bucket(bucketName);
+    const exists = await bucket.exists();
+    if (!exists) {
+        console.error("Bucket does not exist");
+        process.exit(1);
```
Comment (matthewtsmith) [inline google_cloud_storage/index.js]: Use `throw` instead. It automatically sets the exit code correctly

`throw "Bucket does not exist"`

File: google_cloud_storage/index.js [hunk 2]
```diff
@@ -23,5 +28,42 @@ async function uploadFile(input) {
...
+    }
     return {"Success" : true };
+}
+
+async function downloadFile(input) {
+    apptree.validateInputs('Credential', 'FileName', 'ProjectId', 'Bucket', 'ServerFilePath');
+    let credential = input['Credential'];
+    let fileName = input['FileName'];
+    const bucketName = input['Bucket'];
+    const serverDir = input['ServerFilePath'];
```
Comment (matthewtsmith) [inline google_cloud_storage/index.js]: Change it to just `FilePath`. Remove `FileName`

ie.

FilePath: "/users/Matthew/desktop/test.txt"

File: google_cloud_storage/index.js [hunk 3]
```diff
@@ -23,5 +28,42 @@ async function uploadFile(input) {
...
+    const exists = await bucket.exists();
+    if (!exists) {
+        console.error("Bucket does not exist");
+        process.exit(1);
+    }
+
+    try {
+        await bucket.file(fileName).download({destination: `${serverDir}/${fileName}`});
+    } catch (e) {
+        console.error(`error saving file: ${e.toString()}`);
```
Comment (matthewtsmith) [inline google_cloud_storage/index.js]: Use throw

File: google_cloud_storage/package.yaml [hunk 4]
```diff
@@ -25,6 +25,30 @@ steps:
...
+    outputs:
+      Success:
+        description: A boolean indicating success or failure
+  download_file@1.0:
+    inputs:
+      Credentials:
+        required: true
+        description: Your google cloud service account
+        sample: '{"key" : "value"}'
+      FileNath:
```
Comment (matthewtsmith) [inline google_cloud_storage/package.yaml]: Spelling error

**Review by matthewtsmith — CHANGES_REQUESTED** [review]

File: google_cloud_storage/index.js (hunk 1 above)
Comment (alexisandreason) [reply google_cloud_storage/index.js]: updated

File: google_cloud_storage/index.js (hunk 2 above)
Comment (alexisandreason) [reply google_cloud_storage/index.js]: updated

File: google_cloud_storage/index.js (hunk 3 above)
Comment (alexisandreason) [reply google_cloud_storage/index.js]: removed

File: google_cloud_storage/package.yaml (hunk 4 above)
Comment (alexisandreason) [reply google_cloud_storage/package.yaml]: updated

**Pull request merged**